packages/@aws-cdk/aws-ecr/lib/repository-ref.ts

```diff
@@ -2,16 +2,100 @@ import iam = require('@aws-cdk/aws-iam');
 import cdk = require('@aws-cdk/cdk');
 
 /**
- * An ECR repository
+ * Represents an ECR repository.
  */
-export abstract class RepositoryRef extends cdk.Construct {
+export interface IRepository {
+    /**
+     * The name of the repository
+     */
+    readonly repositoryName: string;
+
+    /**
+     * The ARN of the repository
+     */
+    readonly repositoryArn: string;
+
+    /**
+     * The URI of this repository (represents the latest image):
+     *
+     *    ACCOUNT.dkr.ecr.REGION.amazonaws.com/REPOSITORY
+     *
+     */
+    readonly repositoryUri: string;
+
+    /**
+     * Returns the URI of the repository for a certain tag. Can be used in `docker push/pull`.
+     *
+     *    ACCOUNT.dkr.ecr.REGION.amazonaws.com/REPOSITORY[:TAG]
+     *
+     * @param tag Image tag to use (tools usually default to "latest" if omitted)
+     */
+    repositoryUriForTag(tag?: string): string;
+
+    /**
+     * Add a policy statement to the repository's resource policy
+     */
+    addToResourcePolicy(statement: iam.PolicyStatement): void;
+
+    /**
+     * Grant the given principal identity permissions to perform the actions on this repository
+     */
+    grant(identity?: iam.IPrincipal, ...actions: string[]): void;
+
+    /**
+     * Grant the given identity permissions to pull images in this repository.
+     */
+    grantPull(identity?: iam.IPrincipal): void;
+
+    /**
+     * Grant the given identity permissions to pull and push images to this repository.
+     */
+    grantPullPush(identity?: iam.IPrincipal): void;
+}
+
+export interface ImportRepositoryProps {
+  /**
+   * The ARN of the repository to import.
+   *
+   * If you only have a repository name and the repository is in the same account/region
+   * as the current stack, you can use the static method `Repository.arnForLocalRepository(name)`
```

Comment:
Can't we just make ARN optional then, and default it to `Repository.arnForLocalRepository(name)` if it wasn't provided (of course, one of ARN-name would be required)?